tests/autoblocks/test_run_test_suite.py

```diff
@@ -604,6 +604,224 @@ def evaluate_test_case(self, test_case: MyTestCase, output: str) -> Evaluation:
     ]
 
 
+def test_concurrency_dict(httpx_mock):
+    httpx_mock.add_response()
+
+    def test_fn(test_case: MyTestCase):
+        return test_case.input + "!"
+
+    class EvaluatorA(BaseTestEvaluator):
+        id = "evaluator-a"
+
+        def evaluate_test_case(self, test_case: MyTestCase, output: str) -> Evaluation:
+            return Evaluation(score=0)
+
+    class EvaluatorB(BaseTestEvaluator):
+        id = "evaluator-b"
+
+        def evaluate_test_case(self, test_case: MyTestCase, output: str) -> Evaluation:
+            return Evaluation(score=1)
+
+    run_test_suite(
+        id="my-test-id",
+        test_cases=[
+            MyTestCase(input="a"),
+            MyTestCase(input="b"),
+        ],
+        evaluators=[
+            EvaluatorA(),
+            EvaluatorB(),
+        ],
+        fn=test_fn,
+        max_test_case_concurrency=1,
+        max_evaluator_concurrency={
+            "evaluator-a": 1,
+            "evaluator-b": 1,
+        },
```

Comment:
@adamnolte this is what i'm thinking for per-evaluator concurrency

Reply:
Can we statically reference the id's on the evaluators? So like EvaluatorA.id ?

Reply:
yeah good call

Reply:
we could also make it a property of the evaluators themselves:

```python
class EvaluatorA(BaseTestEvaluator):
    max_concurrency = 5
```

Reply:
that would probably make more sense since it would be awkward to have to repeat yourself across multiple test suites if you're sharing an evaluator

Reply:
ya was thinking about that, I think i actually prefer that. @dmorton2297 wdyt?